packages/ui/src/ui-component/dialog/SpeechToTextDialog.js

```diff
@@ -0,0 +1,345 @@
+                        </Select>
+                    </FormControl>
+                </Box>
+                {selectedProvider !== 'none' && (
+                    <>
+                        <ListItem style={{ padding: 0, margin: 0 }} alignItems='center'>
+                            <ListItemAvatar>
+                                <div
+                                    style={{
+                                        width: 50,
+                                        height: 50,
+                                        borderRadius: '50%',
+                                        backgroundColor: 'white'
+                                    }}
+                                >
+                                    <img
+                                        style={{
+                                            width: '100%',
+                                            height: '100%',
+                                            padding: 10,
+                                            objectFit: 'contain'
+                                        }}
+                                        alt='AI'
+                                        src={speechToTextProviders[selectedProvider].icon}
+                                    />
+                                </div>
+                            </ListItemAvatar>
+                            <ListItemText
+                                sx={{ ml: 1 }}
+                                primary={speechToTextProviders[selectedProvider].label}
+                                secondary={
+                                    <a target='_blank' rel='noreferrer' href={speechToTextProviders[selectedProvider].url}>
+                                        {speechToTextProviders[selectedProvider].url}
+                                    </a>
+                                }
+                            />
+                        </ListItem>
+                        {speechToTextProviders[selectedProvider].inputs.map((inputParam, index) => (
+                            <Box key={index} sx={{ p: 2 }}>
+                                <div style={{ display: 'flex', flexDirection: 'row' }}>
+                                    <Typography>
+                                        {inputParam.label}
+                                        {!inputParam.optional && <span style={{ color: 'red' }}>&nbsp;*</span>}
+                                        {inputParam.description && (
+                                            <TooltipWithParser style={{ marginLeft: 10 }} title={inputParam.description} />
+                                        )}
+                                    </Typography>
+                                </div>
+                                {inputParam.type === 'credential' && (
+                                    <CredentialInputHandler
+                                        key={speechToText[selectedProvider]?.credentialId}
```

Comment:
@0xi4o this is where you can specify a `key` so that a new element is rendered, hence you dont need to put `credentialNames` in the `useEffect`